packages/extension-api/src/extension-api.d.ts

```diff
@@ -2425,4 +2425,28 @@ declare module '@podman-desktop/api' {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     export function setValue(key: string, value: any, scope?: 'onboarding'): void;
   }
+
+  export type NotificationType = 'info' | 'warn' | 'error';
+
+  export interface NotificationInfo {
+    extensionId: string;
+    // title displayed on the top of the notification card
+    title: string;
+    // description displayed just below the title, it should explain what the notification is about
+    description: string;
+    type: NotificationType;
+    // displayed below the description, centered in the notification card. It may contains actions (like commands/buttons and links)
+    actions?: string;
+  }
+
+  /**
+   * The context provides write access to the Podman Desktop Notifications system.
+   */
+  export namespace notifications {
+    /**
+     * it pushes a notification that will be displayed in the dashboard/frontend
+     * @param notificationInfo notification to push
+     */
+    export function pushNotification(notificationInfo: NotificationInfo): void;
+  }
```

Comment:
I have mitigated feeling here as we do have the `window.showNotification` method 

so I would expect to reuse `window.showNotification` existing stuff

Reply:
I think using OS notification is not what was designed (looks like a quick implementation at the time it was implemented Also it mostly only works on Windows.

In fact it should be mapped to the notifications / tasks area where you can be notified inside Podman Desktop

Then, what is the difference between "Podman needs to resume setup" and "You have finished to build an image" notifications. it's still a notification so you should see "Podman needs to resume setup in the list of notifications using the bell icon" And be able to resume from there.

If I want to display a notification only in the dashboard then it should be a dedicated type in the option but I think it'll confusing it we proceed using two notification systems in parallel.

Also using Operating System or not should be an implementation details. As a extension author you ask for a thing with option, but it does not contains too much specifics

Going back to the option, it should be like a "highlight" option so it means it'll be hightlighted
Then an implementation of the "highlight" will display it inside the dashboard. But later, let say we discover it should be move to another place, it doesn't change the client code)

so I would use the showNotification with a custom option to highlight/promote this notification

We shouldn't provide any `extensionId: string` attribute. the extension id is provided by the caller. (if I call showNotification I have a custom API that knows from which extension I'm calling then I can call the method using the extensionId (but extension ID is not exposed to the public API, it's only used in the local/implementation)

Reply:
I'm still struggling on exactly what I think the UI/API should look like, but I was about to post something similar. I think the difference is between just briefly telling you about something that happened (e.g. image built successfully), something that you need to acknowledge/dismiss (e.g. maybe a new version was installed and we tell you about new features) and something that you can take action on or dismiss (e.g. click here to setup podman). I don't think these are fundamentally that different, and having a single API will make it easier for us to decide where/how to display them, or change UI as necessary.

Reply:
Updated to "re-use" the old method. It still accepts the old options object to not break the old code.
Is this what you meant? @benoitf 

So, the bell icon list will eventually be fed by a queue with both tasks and notifications, correct?